R/summarize.R

```diff
@@ -0,0 +1,84 @@
+#' summarize WQ site information
+#'
+#' @return data frame of the following information:
+#' @export
+#' @param sites character ID(s) of sites to summarize
+#' @param siteInfo data frame containing site information, with column named CODIGO_ESTACAO 
+#' containing station id's
+#' @param nutriDF data frame record of nutrient measurements
+#' @importFrom dplyr filter 
+#' @importFrom dplyr mutate
+#' @importFrom dplyr bind_cols
+summarizeSites <- function(sites, siteInfo, nutriDF){
+  #get existing site metadata
+  summaryDF <- filter(siteInfo, CODIGO_ESTACAO %in% sites)
+  
+  #add metrics that have to be calculated
+  if(!is.Date(nutriDF$date)){
+    nutriDF <- mutate(nutriDF, date = as.Date(date))
+  }
+  
+  #will need work if summarizeSites will deal with multiple sites
+  #could be useful as a stand-alone function?
+  dateStats <- getDateStats(nutriDF$date)
+  
+  summaryDF <- bind_cols(summaryDF, dateStats)
+  
+  return(summaryDF)
+}
+
+#' Get various summary statistics on a date vector
+#'
+#' @param dateCol vector of dates
+#' 
+#' @return data frame of date statistics, including 
+#'
+getDateStats <- function(dateCol){
+  #is there a function that could be used instead of this?
+  start <- min(dateCol)
+  end <- max(dateCol)
+  n <- length(dateCol)
+  statDF <- data.frame(start, end, n, stringsAsFactors = FALSE)
+  return(statDF)
+}
+
+#' Extract various model summary statistics
+#' 
+#' @param model the input model object; will extend to several types
+#' @export
+summarizeModel <- function(model) UseMethod("summarizeModel")
+
+#' stats for rloadest loadReg2 model
+#' @rdname summarizeModel
+#' 
+summarizeModel.loadReg2 <- function(model){
+  
+}
+
+#' Summarize generated predictions 
+#' just take one site for now
+#'
+#' @param preds data frame input data frame of predicted solutes
+#' @param meta loadflex metadata object for the preds data frame
+#' @param by character "total" to return average flux over all years, "annual" to return 
+#' yearly (water year) averages
+#' @param modelName char name of model used to generate predictions
+#' @return data frame containing various statistics
+#' @export
+#'
+summarizePreds <- function(preds, meta, by, modelName){
+   station <- getInfo(meta, field = c("station"))
+   if(by == "total"){
+    annuals <- aggregateSolute(preds, metadata = meta, format = "flux rate",
```

Comment:
this function comes pretty close to just reproducing the functionality of `aggregateSolute`. let's consider having this function accept pre-aggregated predictions instead, or passing in more of the arguments to aggregateSolute more directly (e.g., pass dots arg from summarizePreds into aggregateSolute to cover stuff like `format` and `agg.by`, or something.